routers/web/user/home.go

```diff
@@ -454,10 +454,21 @@ func buildIssueOverview(ctx *context.Context, unitType unit.Type) {
 		AllPublic:  false,
 		AllLimited: false,
 	}
-
 	if team != nil {
 		repoOpts.TeamID = team.ID
 	}
+	{
+		ids, _, err := repo_model.SearchRepositoryIDs(repoOpts)
+		if err != nil {
+			ctx.ServerError("SearchRepositoryIDs", err)
+			return
+		}
+		opts.RepoIDs = ids
+		if len(opts.RepoIDs) == 0 {
+			// no repos found, don't let the indexer return all repos
+			opts.RepoIDs = []int64{0}
+		}
+	}
```

Comment:
Curious, what's this `{}` syntax? Is this a on-demand closure?

Reply:
It's an `int64` slice with a single value `0` in it.

Reply:
Found it, it's called Block: https://go.dev/ref/spec#Blocks

Reply:
Ah, my bad! Yep, a block.

Reply:
Never mind, it is just a trick. I usually do this in a relatively separate piece of code, to avoid polluting a broader scope with local variables, especially when the function body is large.

`buildIssueOverview` is so big, it has almost 400 lines and at least 20 local variables.

Explicit blocks also make it clear how to refactor them to multiple functions, but I'm too lazy to refactor it right now 😁